streaming/index.js

```diff
@@ -183,6 +184,73 @@ const startServer = async () => {
   const redisSubscribeClient = await redisUrlToClient(redisParams, redisUrl);
   const redisClient = await redisUrlToClient(redisParams, redisUrl);
 
+  // Collect metrics from Node.js
+  metrics.collectDefaultMetrics();
+
+  new metrics.Gauge({
+    name: 'pg_pool_total_connections',
+    help: 'The total number of clients existing within the pool',
+    collect() {
+      this.set(pgPool.totalCount);
+    },
+  });
+
+  new metrics.Gauge({
+    name: 'pg_pool_idle_connections',
+    help: 'The number of clients which are not checked out but are currently idle in the pool',
+    collect() {
+      this.set(pgPool.idleCount);
+    },
+  });
+
+  new metrics.Gauge({
+    name: 'pg_pool_waiting_queries',
+    help: 'The number of queued requests waiting on a client when all clients are checked out',
+    collect() {
+      this.set(pgPool.waitingCount);
+    },
+  });
+
+  const connectedClients = new metrics.Gauge({
+    name: 'connected_clients',
+    help: 'The number of clients connected to the streaming server',
+    labelNames: ['type'],
+  });
+
+  connectedClients.set({ type: 'websocket' }, 0);
+  connectedClients.set({ type: 'eventsource' }, 0);
+
+  const connectedChannels = new metrics.Gauge({
+    name: 'connected_channels',
+    help: 'The number of channels the streaming server is streaming to',
+    labelNames: [ 'type', 'channel' ]
+  });
+
+  const redisSubscriptions = new metrics.Gauge({
+    name: 'redis_subscriptions',
+    help: 'The number of Redis channels the streaming server is subscribed to',
+  });
+
+  // When checking metrics in the browser, the favicon is requested this
+  // prevents the request from falling through to the API Router, which would
+  // error for this endpoint:
+  app.get('/favicon.ico', (req, res) => res.end());
+
+  app.get('/api/v1/streaming/health', (req, res) => {
```

Comment:
Ideally this should be changed to `/health`, but that could break people's monitoring & proxying setups, so I've not made that change for now.

Reply:
We want this endpoint to be accessible from the end-user, and when the streaming server is on the main domain, it means it needs to be reachable under `/api/v1/streaming/`

Reply:
Why does a health check endpoint need to be publicly accessible? It's only used by monitoring..?

Reply:
I think the intent was at some point that apps could check it to know if it was working correctly, rather than attempting to retry a WS connection

Reply:
I don't think this makes sense, especially in the load balanced environment: say you run multiple instances of streaming, and have nginx load balance them, then the `GET /api/v1/streaming/health` could hit the healthy instance, an the WS connection could reach an unhealthy instance.

The correct strategy here is exponential reconnection backoff with initial jitter and max retries before failing to non-streaming connectivity.

Reply:
I agree, but as I remember this was the intent behind making this public.

Changing this may be a breaking change, I dont know if this is used by some clients.

Reply:
Yeah, so, I'm leaving it as a public endpoint now, but I think it'd be very wise to announce that it will be deprecated in 4.3.0 or something — giving people time to upgrade away from it, if they're using it.

Reply:
Then we also want to have `/health` answering the same response as this one, and people can move to it before its deprecated

Reply:
I could add in `/health` now, just wasn't sure if I'd get approval overall for that; hence just pointing it out now that it's technically an internal endpoint, not an external one